File: public/dsa/arrays/largest_contiguous_sum/kadane_variation/largest_contiguous_sum.py


from typing import List


class Solution:
    def largestContiguousSum(self, arr: List[int]) -> int:
        cur_sum = arr[0]
        max_sum = cur_sum
        for i in range(1, len(arr)):
            if cur_sum < 0:
                cur_sum = 0
            cur_sum += arr[i]
            max_sum = max(max_sum, cur_sum)
        return max_sum


# tests-start
arr = [4, -6, 2, 5]
sol = Solution()
print(sol.largestContiguousSum(arr))
# tests-end

'''output
7
'''

File: public/dsa/arrays/largest_contiguous_sum/kadane_variation/largest_contiguous_sum.cpp
#include <bits/stdc++.h>
using namespace std;

int largestContiguousSum(vector<int> &arr) {
    int cur_sum = arr[0];
    int max_sum = cur_sum;

    for (int i = 1; i < arr.size(); i++) {
        if (cur_sum < 0) {
            cur_sum = 0;
        }
        cur_sum += arr[i];
        max_sum = max(max_sum, cur_sum);
    }
    return max_sum;
}

// tests-start
int main() {
    vector<int> arr = {4, -6, 2, 5};
    cout << largestContiguousSum(arr) << endl;
    return 0;
}
// tests-end

/*output
7
*/